### PR TITLE
BE | Ask VA Api: fix `0873.rb` error handling bug

### PR DESCRIPTION
**Pull request by Khoa-V-Nguyen**
## Summary

- Fix `preferred_name` and `ava_profile` method in `0873.rb`

## Related issue(s)

- https://github.com/department-of-veterans-affairs/ask-va/issues/1464
## Testing done

- [x] *New code is covered by unit tests*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

**Review by joehall-tw — DISMISSED** [review]
LGTM...

File: app/models/form_profiles/va_0873.rb [hunk 1]
```diff
@@ -42,11 +42,19 @@ def initialize_personal_information
...
 
   def initialize_ava_profile
-    school = GIDSRedis.new.get_institution_details_v0(id: profile.school_facility_code)
-    school_name = school[:data][:attributes][:name]
+    school_name = nil
+
+    unless profile.school_facility_code.nil?
+      school = GIDSRedis.new.get_institution_details_v0(id: profile.school_facility_code)
+      school_name = school[:data][:attributes][:name]
+    end
```
Comment (rmtolmach) [inline app/models/form_profiles/va_0873.rb:56]: Can you add a test to `form_profile_spec`? There is a test there now that uses this value: `'schoolFacilityCode' => '12345678'`, so you could do one with `nil` as well.

Comment (Khoa-V-Nguyen) [reply app/models/form_profiles/va_0873.rb:56]: Updated.

File: spec/models/form_profile_spec.rb [hunk 2]
```diff
@@ -1663,11 +1663,56 @@ def expect_prefilled(form_id)
...
+        context 'when school facility code is nil' do
+          let(:info) do
+            {
+              SchoolFacilityCode: nil,
+              BusinessPhone: '1234567890',
+              BusinessEmail: 'fake@company.com',
+              ServiceNumber: '123455678'
+            }
+          end
+          let(:v0873_expected) do
```
Comment (rmtolmach) [inline spec/models/form_profile_spec.rb:1686]: There already is one of these blocks defined  on line 111. Is there a reason you didn't update that one?

Comment (Khoa-V-Nguyen) [reply spec/models/form_profile_spec.rb:1686]: because i have two test, one test is for when all the data is there. And the other one is for when school_facility_code is nil

File: spec/models/form_profile_spec.rb [hunk 3]
```diff
@@ -1663,11 +1663,56 @@ def expect_prefilled(form_id)
...
+              },
+              'veteranServiceInformation' => veteran_service_information
+            }
+          end
+
+          it 'does not show in ava profile' do
+            VCR.use_cassette('va_profile/demographics/demographics', VCR::MATCH_EVERYTHING) do
+              VCR.use_cassette('va_profile/military_personnel/post_read_service_histories_200',
+                               allow_playback_repeats: true, match_requests_on: %i[uri method body]) do
+                expect_prefilled('0873')
```
[the remark this answers is not in the document]
Comment (Khoa-V-Nguyen) [reply spec/models/form_profile_spec.rb:1714]: schoolInfo is no longer there.

     let(:v0873_expected) do
            {
              'personalInformation' => {
                'first' => user.first_name&.capitalize,
                'last' => user.last_name&.capitalize,
                'suffix' => user.suffix,
                'preferredName' => 'SAM',
                'dateOfBirth' => user.birth_date,
                'socialSecurityNumber' => user.ssn,
                'serviceNumber' => '123455678'
              },
              'contactInformation' => {
                'email' => user.pciu_email,
                'phone' => us_phone,
                'address' => address
              },
              'avaProfile' => {
                'businessPhone' => '1234567890',
                'businessEmail' => 'fake@company.com'
              },
              'veteranServiceInformation' => veteran_service_information
            }
          end

[the remark this answers is not in the document]
Comment (rmtolmach) [reply spec/models/form_profile_spec.rb:1714]: oh, i see 👍

File: spec/models/form_profile_spec.rb [hunk 4]
```diff
@@ -1663,11 +1663,56 @@ def expect_prefilled(form_id)
...
+        end
+
+        context 'when school facility code is nil' do
+          let(:info) do
+            {
+              SchoolFacilityCode: nil,
+              BusinessPhone: '1234567890',
+              BusinessEmail: 'fake@company.com',
+              ServiceNumber: '123455678'
+            }
```
Comment (rmtolmach) [inline spec/models/form_profile_spec.rb:1684]: Is`info` being used anywhere?

Comment (Khoa-V-Nguyen) [reply spec/models/form_profile_spec.rb:1684]: here:

      context 'with VA Profile prefill for 0873' do
        let(:info) do
          {
            SchoolFacilityCode: '12345678',
            BusinessPhone: '1234567890',
            BusinessEmail: 'fake@company.com',
            ServiceNumber: '123455678'
          }
        end
        let(:profile) do
          AskVAApi::Profile::Entity.new(info)
        end

Comment (rmtolmach) [reply spec/models/form_profile_spec.rb:1684]: oh, okay, so it's needed when it's instantiated. Thanks for the clarification!

**Review by rmtolmach — APPROVED** [review]
